crypto/src/asset/denom.rs

```diff
@@ -1,360 +1,25 @@
-    pub(super) unit_index: usize,
-}
-
-// These are constructed by the asset registry.
-pub(super) struct Inner {
-    id: asset::Id,
-    base_denom: String,
-    /// Sorted by priority order.
-    pub(super) units: Vec<UnitData>,
-}
-
-pub(super) struct UnitData {
-    pub exponent: u8,
-    pub denom: String,
-}
-
-impl Inner {
-    /// Constructs the backing data for a set of units.
-    ///
-    /// The base denom is added as a unit, so `units` can be empty and should
-    /// not include a unit for the base denomination.
-    pub fn new(base_denom: String, mut units: Vec<UnitData>) -> Self {
-        let id = asset::Id(Fq::from_le_bytes_mod_order(
-            // XXX choice of hash function?
-            blake2b_simd::Params::default()
-                .personal(b"Penumbra_AssetID")
-                .hash(base_denom.as_bytes())
-                .as_bytes(),
-        ));
-
-        for unit in &units {
-            assert_ne!(unit.exponent, 0);
-            assert_ne!(&unit.denom, &base_denom);
-        }
-
-        units.push(UnitData {
-            exponent: 0,
-            denom: base_denom.clone(),
-        });
-
-        Self {
-            id,
-            units,
-            base_denom,
-        }
-    }
-}
-
-impl Denom {
-    /// Return the [`asset::Id`] associated with this denomination.
-    pub fn id(&self) -> asset::Id {
-        self.inner.id.clone()
-    }
-
-    /// Create a value of this denomination.
-    pub fn value(&self, amount: asset::Amount) -> Value {
-        Value {
-            amount,
-            asset_id: self.id(),
-        }
-    }
-
-    /// Return a list of display units for this denomination, in size order.
-    ///
-    /// There will always be at least one display denomination.
-    pub fn units(&self) -> Vec<Unit> {
-        (0..self.inner.units.len())
-            .map(|unit_index| Unit {
-                unit_index,
-                inner: self.inner.clone(),
-            })
-            .collect()
-    }
-
-    /// Returns the default (largest) unit for this denomination.
-    pub fn default_unit(&self) -> Unit {
-        Unit {
-            unit_index: 0,
-            inner: self.inner.clone(),
-        }
-    }
-
-    /// Returns the base (smallest) unit for this denomination.
-    ///
-    /// (This treats the base denomination as a display unit).
-    pub fn base_unit(&self) -> Unit {
-        Unit {
-            unit_index: self.inner.units.len() - 1,
-            inner: self.inner.clone(),
-        }
-    }
-
-    /// Returns the "best" unit for the given amount (expressed in units of the
-    /// base denomination).
-    ///
-    /// This is defined as the largest unit smaller than the given value (so it
-    /// has no leading zeros when formatted).
-    pub fn best_unit_for(&self, amount: asset::Amount) -> Unit {
-        // Special case: use the default unit for 0
-        if amount == 0u64.into() {
-            return self.default_unit();
-        }
```

Comment:
Here's the special case that seems to be missing in the new code

Reply:
Oh, that makes sense! All changes to `denom.rs` will not be taken into account on this branch on account of the renaming of that file/struct & creation of a new one